lib/Dialect/FIRRTL/FIRRTLOpInterfaces.cpp

```diff
@@ -81,9 +81,10 @@ LogicalResult circt::firrtl::verifyModuleLikeOpInterface(FModuleLike module) {
     return module.emitOpError("requires valid port symbols");
   if (!portSymbols.empty() && portSymbols.size() != numPorts)
     return module.emitOpError("requires ") << numPorts << " port symbols";
-  if (llvm::any_of(portSymbols.getValue(),
-                   [](Attribute attr) { return !attr.isa<StringAttr>(); }))
-    return module.emitOpError("port symbols should all be string attributes");
+  if (llvm::any_of(portSymbols.getValue(), [](Attribute attr) {
+        return !attr || !attr.isa<InnerSymAttr>();
+      }))
```

Comment:
Maybe it would be nice to check `attr.cast<InnerSymAttr>().isValid`(`!attr.cast<InnerSymAttr>().empty()`).

Reply:
Actually, we have empty or invalid entries in the `portSymbols` now. It is required to store ports with no symbol in the `portSymbols` array.